compiler-rt/lib/hwasan/hwasan_report.cpp

```diff
@@ -446,8 +458,51 @@ class BaseReport {
   const HeapChunk heap;
   const Allocations allocations;
   const OverflowCandidate candidate;
+
+  struct {
+    uptr addr = 0;
+    tag_t tags[512] = {};
```

Comment:
should this be odd so we have the same amount of tags on both sides of the address?

Reply:
we will align on 16, so we want the same number of lines before and after (17lines x 16bytes)
so logic will be more complicated, but possible.
However if we want to keep printing outside (we may want a print tags function outside of reporting), then I'd rather go with larger snapshot, but simpler logic.
 
Another issue is that when we do a snapshot we don't use a pierce address. in one of usecases we add access size as offset.